batchflow/models/tf/efficientnet.py

```diff
@@ -0,0 +1,318 @@
+""" Base class for scalable models and EfficientNet
+
+Mingxing Tan, Quoc V. Le "`EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks
+<https://arxiv.org/abs/1905.11946>`_"
+
+"""
+import numpy as np
+import tensorflow as tf
+
+from . import TFModel, MobileNet_v2
+from .layers import conv_block
+
+
+class ScalableModel(TFModel):
```

Comment:
Overall, that looks great. My only concern is that you use `factors` to modify different parameters right before their application. Why not just scan the whole `config` once and update all the values?

Just a quick thought: it is possible to make `@scalable` decorator and apply that to any of the existing class. Under the hood, that decorator would overload `build_config` method by:

- calling the `build_config` (that is, from the decorated class)

- applying the scaling logic to the confiuration-dictionary

Obviously, there are some questions (for example, what should be the right place for `factor's` in this case).

Reply:
I will consider modifying config in build_config directly, but as for now we do need body to iterate over blocks and to process `repeats` properly if we use something other than MobileNet_v2.block